lib/pages/home/survey_page_viewer.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_survey/gen/assets.gen.dart';
+import 'package:flutter_survey/pages/widgets/dimmed_background.dart';
+
+class SurveyPageViewer extends StatelessWidget {
+  final List<int> surveys;
+  final ValueNotifier<int> currentPageNotifier;
+  final _pageController = PageController();
+
+  SurveyPageViewer({required this.surveys, required this.currentPageNotifier});
+
+  @override
+  Widget build(BuildContext context) {
+    return PageView.builder(
+      itemCount: surveys.length,
+      controller: _pageController,
+      itemBuilder: (BuildContext context, int index) {
+        return _buildSurveyPageView(context);
+      },
+      onPageChanged: (int index) {
+        currentPageNotifier.value = index;
+      },
+    );
+  }
+
+  Widget _buildSurveyPageView(BuildContext context) {
+    return Stack(
+      children: [
+        Container(
+          decoration: BoxDecoration(
+            image: DecorationImage(
+              image: Assets.images.bgHomeSample,
+              fit: BoxFit.cover,
+            ),
+          ),
+        ),
+        _buildDimmedBackground(),
+      ],
+    );
+  }
+
+  Widget _buildDimmedBackground() => DimmedBackground(
```

Comment:
I think as you had already created a DimmedBackground widget, maybe just stick it directly to the tree of widget instead of adding another function to interrupt the reading flow 🐧

Reply:
Fixed https://github.com/luongvo/flutter-survey/pull/49/commits/e10b879e4ee80a410b57ebb4d544dbeee7314d4b